Sources/ComposableArchitecture/SharedState/PersistenceKey/AppStorageKey.swift

```diff
@@ -154,7 +154,7 @@ extension PersistenceReaderKey {
 /// A type defining a user defaults persistence strategy.
 ///
 /// See ``PersistenceReaderKey/appStorage(_:)-4l5b`` to create values of this type.
-public struct AppStorageKey<Value> {
+public struct AppStorageKey<Value: Sendable>: Sendable {
   private let lookup: any Lookup<Value>
   private let key: String
   private let store: UserDefaults
```

Comment:
Technically `UserDefaults` is not `Sendable` even though the docs specifically say "User defaults is thread-safe". So, it should be ok to make this `UncheckedSendable`, but it was strangely causing some test failures. So I need to look at this again.